ppoprf/src/ppoprf.rs

```diff
@@ -68,6 +70,52 @@ impl ProofDLEQ {
     Self { c: chl, s }
   }
 
+  fn new_batch(
+    key: &RistrettoScalar,         //k
+    public_value: &RistrettoPoint, //Y -> B
+    p: &[RistrettoPoint],          //C
+    q: &[RistrettoPoint],          //D
+  ) -> Self {
+    //(M, Z) = ComputeCompositesFast(k, B, C, D)
+    let (m, z) = ProofDLEQ::compute_composites(Some(*key), public_value, p, q);
```

Comment:
As this is the fast version, it needs to be enforced that the key is passed.

Reply:
How would I enforce that the key is used here? I would assume a compile error when the key is None.

Reply:
Enforcing passing the key would just be passing it directly, without the `Option` wrapper. But I don't know what that means for the verify step. @claucece could you clarify here, please?

Reply:
I added the optional key parameter exactly so that I can differentiate between the computation for M, Z for either the `new` method which knows the key and therefore can faster calculated M, Z, and the `verify` method, that doesn't know the key and therefore needs to calculate M, Z differently. So I use the key to differentiated between ComputeComposites and ComputeCompositesFast.

Reply:
Looks good now!